### DISCUSSION TITLE
Is it possible to leverage a repeated pattern as a custom entity?

### DISCUSSION THREAD

**Discussion by timturnerwhcc** [winkjs/wink-nlp#69] [Help on coding] [answered]
The question is ambiguous, so let me provide an example. I, effectively, want to leverage the `+` regex operator around a custom entity pattern to understand if I'm looking at a list of things, rather than each thing within a list.

**Example**:
Input Document: "Tom, Dana, James, Lily, and Ronan"

**Custom Entity:**
`{ name: 'firstNameList', patterns: ['[PROPN] [|PUNCT|CCONJ]'], mark: [0, 0] }`

The above custom entity will detect each proper noun as a separate custom entity. I would like a single entity that represents the entire list, which can have an arbitrary length. Is this possible?

Thank you!

**Comment by sanjayaksaxena** [comment, marked as answer]
Currently `+` regex operator like operation is not supported; it is there in our ToDos but so far we have not been able to devote time on this.

As a work around, you may like to try following code:

```javascript
const winkNLP = require( 'wink-nlp' );
const its = require( 'wink-nlp/src/its.js' );
const model = require( 'wink-eng-lite-web-model' );
const nlp = winkNLP( model );
const p = [
  { name: 'firstNameList', patterns: ['[PROPN] [|PUNCT|CCONJ] [|CCONJ]'] }
];
nlp.learnCustomEntities( p );
const text = 'Tom, Dana, James, Lily, and Ronan went to park. They met John and Mike in the park.';
const doc = nlp.readDoc( text );
doc.sentences().each( ( s ) => console.log( s.customEntities().out().join( ' ' ) ) )
// -> "Tom, Dana, James, Lily, and Ronan"
// -> "John and Mike"
```

Note, the above code will work only if there is a single list per sentence!

**Reply by timturnerwhcc** [reply]
Ah, excellent, thank you! That should work for my use case.
